compositor: Have viewport meta tag affect the layout viewport

atbrakhi opened the pull request:
The CSS specifications defines two viewports[^1], the "layout viewport" and the visual viewport. The layout viewport affects layout and the visual viewport is the result of applying pinch zoom. This changes makes it so that the initial-scale property of `<meta type="viewport">` affects the layout viewport.

Testing: Added a new unit test

[^1]: https://www.w3.org/TR/css-viewport-1/#the-viewport

mrobinson commented:
> Isn't it what I did here : #39067 ? @mrobinson
> 
> cc: @xiaochengh

I left a comment asking you to reopen the PR if it was still an issue (as the original bug was closed), but didn't see any activity. We were reading the specifications and stumbled on this issue independently. We'll add you as a co-author here.

atbrakhi committed:
Have viewport meta tag affect the layout viewport

The CSS specifications defines two viewports, the "layout viewport" and the
visual viewport. The layout viewport affects layout and the visual viewport is
the result of applying pinch zoom. This changes makes it so that the
initial-scale property of <meta type="viewport"> affects the layout viewport.

Co-authored-by: Shubham Gupta <shubham13297@gmail.com>
Co-authored-by: Martin Robinson <mrobinson@igalia.com>
Signed-off-by: atbrakhi <atbrakhi@igalia.com>
